src/form_helper.js

```diff
@@ -256,36 +245,41 @@ export default class FormHelper {
   }
 
   _setElementValue(element, value, elementName){
-    if (element) {
-      element.value = value;
+    if (!element) {
+      this.logErrorMessage(element, value, elementName)
+   }
 
-      var event = document.createEvent('HTMLEvents');
-      event.initEvent('change', true, false);
-      element.dispatchEvent(event);
+   else if (element.options) {
+      const find = (f, x) => Array.prototype.find.call(x, f)
+      const checkOptionMatchesValue = option => option.value == value
 
-      var options = element.options;
-      if (options) {
-        for (var i = 0; i < options.length; i++) {
-          if (element.options[i].value == value) {
-            element.options[i].dispatchEvent(event);
-            break;
-          }
-        }
-      }
+      const foundElement = find(checkOptionMatchesValue, element.options)
+      if (foundElement) this.createEvent(foundElement)
+    }
 
-      return;
+    else {
+      element.value = value;
+      this.createEvent(element)
     }
+  }
 
+  logErrorMessage() {
     var errorMessage = 'AddressFinder Error: '
                        + 'Attempted to update value for element that could not be found.\n'
                        + '\nElement: ' + elementName
                        + '\nValue: ' + value;
+   if (window.console) {
+     console.warn(errorMessage);
+   }
+  }
 
-    if (window.console) {
-      console.warn(errorMessage);
-    }
+  createEvent(element) {
```

Comment:
I would probably not extract this out into a _public function_, as it's fairly specific to the `setElementValue` function. Let's discuss.